lib/node_modules/@stdlib/fs/resolve-parent-paths-by/README.md

```diff
@@ -0,0 +1,225 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+-->
+
+# resolveParentPathsBy
+
+> Resolve paths from a set of paths according to a predicate function by walking parent directories.
+
+<section class="usage">
+
+## Usage
+
+```javascript
+var resolveParentPathsBy = require( '@stdlib/fs/resolve-parent-paths-by' );
+```
+
+<a name="resolve-parent-paths-by"></a>
+
+#### resolveParentPathsBy( paths\[, options], predicate, clbk )
+
+Asynchronously resolves paths from a set of paths according to a `predicate` function by walking parent directories.
+
+```javascript
+resolveParentPathsBy( [ 'package.json', 'package-lock.json' ], predicate, onPath );
+
+function predicate( path, next ) {
+    var bool = ( /\/test\//.test( path ) === false );
+    next( null, bool );
+}
+
+function onPath( error, path ) {
+    if ( error ) {
+        throw error;
+    }
+    console.log( path );
+    // e.g., => '...'
+}
+```
+
+The function accepts the following `options`:
+
+-   **dir**: base directory from which to begin walking. May be either an absolute path or a path relative to the current working directory.
+
+-   **mode**: mode of operation. The following modes are supported:
+
+    -   `first`: return the first resolved path.
+    -   `some`: return one or more paths resolved within the first directory level containing a match.
+    -   `all`: return all resolved paths within the first directory level containing matches for all provided paths.
+    -   `each`: independently return the first resolved path for each path at any directory level.
+    
+    Default: `'all'`.
+
+By default, the function begins walking from the current working directory. To specify an alternative directory, set the `dir` option.
+
+```javascript
+var opts = {
+    'dir': __dirname
+};
+resolveParentPathsBy( [ 'package.json' ], opts, predicate, onPath );
+
+function predicate( path, next ) {
+    var bool = ( /\/test\//.test( path ) === false );
+    next( null, bool );
+}
+
+function onPath( error, path ) {
+    if ( error ) {
+        throw error;
+    }
+    console.log( path );
+    // e.g., => '...'
+}
+```
+
+When invoked, the `predicate` function is provided two arguments:
+
+-   `path`: a resolved path.
+-   `next`: a callback which should be called once the `predicate` function has finished processing a resolved path.
+
+If a `predicate` function calls the `next` callback with a `truthy` second argument, the function proceeds to next function and returns the resolved path.
+
+If unable to resolve a path, the function returns `null` as the path result.
```

Comment:
The function doesn't return `null`, either an empty array or `Array<null>` in case of `each` mode. I would suggest being specific and moving this to the `## Notes` section. I'll update `resolve-parent-paths` accordingly too. 

cc: @kgryte